extensions/panache/mongodb-panache-kotlin/deployment/src/test/kotlin/io/quarkus/mongodb/panache/kotlin/deployment/TestEnhancers.java

```diff
@@ -0,0 +1,192 @@
+package io.quarkus.mongodb.panache.kotlin.deployment;
+
+import static java.lang.String.format;
+import static java.util.stream.Collectors.toList;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Map;
+import java.util.StringJoiner;
+import java.util.TreeMap;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import org.bson.types.ObjectId;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.objectweb.asm.ClassReader;
+import org.objectweb.asm.tree.ClassNode;
+import org.objectweb.asm.tree.MethodNode;
+import org.objectweb.asm.util.Printer;
+import org.objectweb.asm.util.Textifier;
+import org.objectweb.asm.util.TraceClassVisitor;
+
+import io.quarkus.gizmo.Gizmo;
+import io.quarkus.mongodb.panache.kotlin.PanacheMongoCompanionBase;
+import io.quarkus.mongodb.panache.kotlin.PanacheMongoRepositoryBase;
+import io.quarkus.panache.common.deployment.ByteCodeType;
+import io.quarkus.panache.common.impl.GenerateBridge;
+import io.quarkus.test.QuarkusUnitTest;
+
+@SuppressWarnings("unchecked")
+public class TestEnhancers {
+    public static final ByteCodeType CLASS = new ByteCodeType(Class.class);
+    public static final ByteCodeType OBJECT_ID = new ByteCodeType(ObjectId.class);
+    public static final ByteCodeType LONG = new ByteCodeType(long.class);
+    public static final ByteCodeType GENERATE_BRIDGE = new ByteCodeType(GenerateBridge.class);
+    public static final Pattern LABEL = Pattern.compile("^L(\\d)$");
+    @RegisterExtension
+    static final QuarkusUnitTest config = new QuarkusUnitTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addClasses(Book.class)
+                    .addClasses(Student.class)
+                    .addClasses(StudentRepository.class));
+
+    @Test
+    public void testCompanion() throws Exception {
+        evaluate(new ByteCodeType(Book.Companion.getClass()), new ByteCodeType(Book.class), PanacheMongoCompanionBase.class, OBJECT_ID);
+    }
+
+    @Test
+    public void testRepository() throws Exception {
+        evaluate(new ByteCodeType(StudentRepository.class), new ByteCodeType(Student.class), PanacheMongoRepositoryBase.class, LONG);
+    }
+
+    /**
+     * we don't generate labels, e.g., like the compiler does (it's sourcemap related) so we need to elide some lines
+     * to account for the difference.
+     *
+     * @return
+     */
+    private List<Object> cleanUp(List<Object> input) {
```

Comment:
Can you move this method closer to where it is used ? For ex. after `validate()`.

Reply:
done